File: Day121_2025/Q121_GFG.java
import java.util.*;

class Q121_GFG {

    ArrayList<Integer> nthRowOfPascalTriangle(int n) {
        ArrayList<Integer> res = new ArrayList<>();
        long prev = 1;
        res.add((int) prev);

        for (int i = 1; i < n; i++) {
            long curr = (prev * (n - i)) / i;
            res.add((int) curr);
            prev = curr;
        }

        return res;
    }
}

File: Day121_2025/Q121_GFG.py
class Solution:

    def nthRowOfPascalTriangle(self, n):
        res = []
        prev = 1
        res.append(prev)

        for i in range(1, n):
            curr = (prev * (n - i)) // i
            res.append(curr)
            prev = curr

        return res
